plugin/src/main/groovy/com/novoda/staticanalysis/internal/ktlint/KtlintConfigurator.groovy

```diff
@@ -0,0 +1,97 @@
+package com.novoda.staticanalysis.internal.ktlint
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.Violations
+import com.novoda.staticanalysis.internal.Configurator
+import com.novoda.staticanalysis.internal.VariantFilter
+import com.novoda.staticanalysis.internal.checkstyle.CollectCheckstyleViolationsTask
+import org.gradle.api.GradleException
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+class KtlintConfigurator implements Configurator {
+
+    private static final String KTLINT_PLUGIN = 'org.jlleitschuh.gradle.ktlint'
+    private static final String KTLINT_NOT_APPLIED = 'The Ktlint plugin is configured but not applied. Please apply the plugin in your build script.\nFor more information see https://github.com/jeremymailen/kotlinter-gradle'
+
+    private final Project project
+    private final Violations violations
+    private final Task evaluateViolations
+    private final VariantFilter variantFilter
+
+    static KtlintConfigurator create(Project project,
+                                     NamedDomainObjectContainer<Violations> violationsContainer,
+                                     Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('ktlint')
+        return new KtlintConfigurator(project, violations, evaluateViolations)
+    }
+
+    KtlintConfigurator(Project project, Violations violations, Task evaluateViolations) {
+        this.project = project
+        this.violations = violations
+        this.evaluateViolations = evaluateViolations
+        this.variantFilter = new VariantFilter(project)
+    }
+
+    @Override
+    void execute() {
+        project.extensions.findByType(StaticAnalysisExtension).ext.ktlint = { Closure config ->
+            if (!project.plugins.hasPlugin(KTLINT_PLUGIN)) {
+                throw new GradleException(KTLINT_NOT_APPLIED)
+            }
+
+            def ktlint = project.ktlint
+            ktlint.ignoreFailures = true
+            ktlint.reporters = ['CHECKSTYLE', 'PLAIN']
+            ktlint.ext.includeVariants = { Closure<Boolean> filter ->
+                variantFilter.includeVariantsFilter = filter
+            }
+            config.delegate = ktlint
+            config()
+
+            project.afterEvaluate {
```

Comment:
Just a notice: Had to add `project.afterEvaluate` since the plugin itself uses `afterEvaluate` too. Don't think it is a problem.